### ISSUE TITLE
Automatically adapt Mapbox Streets–sourced layers for Dynamic Type

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-gl-native#7030]
When a user adjusts the Dynamic Type or Accessibility Type setting in the Accessibility section of Settings, any style based on the Mapbox Streets source should automatically adjust by making the labels larger or smaller. We could make this behavior configurable for all layers, regardless of source, by having MGLStyleLayer conform to UIContentSizeCategoryAdjusting.

MGLMapView can observe [`UIContentSizeCategoryDidChangeNotification`](https://developer.apple.com/reference/uikit/uicontentsizecategorydidchangenotification?language=objc) and respond by obtaining [`UIApplication.preferredContentSizeCategory`](https://developer.apple.com/reference/uikit/uiapplication/1623048-preferredcontentsizecategory?language=objc), determining the `identifier`s of the sources whose `URL` matches `mapbox://mapbox.mapbox-streets-v*`, and adjusting the `textSize` of any MGLSymbolStyleLayer whose `sourceIdentifier` is one of those source identifiers.

Iterating over all the sources and style layers will require #6097. Discovering whether a source is Mapbox Streets will require #6584.

/ref tangrams/tangram-es#920
/cc @incanus @pveugen @boundsj @frederoni

**Comment by 1ec5** [comment]
This is an alternative approach to #2429 that I think would be less aggressive.

https://github.com/mapbox/mapbox-gl-native/issues/2429#issuecomment-143821971 points out that shield images would need to be scaled to fit their text. We’d have to work out how the `iconTextFit` property fits in with this proposal.

**Comment by 1ec5** [comment]
#7031 would rely on a similar mechanism. I suspect localizing labels would be less cartographically disruptive than changing their size, but I think both should ultimately be the responsibility of the SDK.

**Comment by kkaefer** [comment]
I don't think we should implement that kind of style manipulation at an SDK level. We need that feature on Android as well and should add it to core.

**Comment by 1ec5** [comment]
That's true; accessibility text sizes aren't as specific to a particular source as languages are. The approach described here would require core changes, anyhow, because UIContentSizeCategoryAdjusting conformance needs to be implemented in a way that persists for the lifetime of an mbgl::style::Layer, not just an MGLStyleLayer that happens to wrap it.

**Comment by felixLam** [comment]
For anyone interested we have open-sourced a "simple" approach to adjusting the fonts to match the user's preferred content size: https://github.com/iosphere/ISHMapboxDynamicFontObserver

**Comment by hyerra** [comment]
Just double checking but is dynamic type functionality integrated into MapBox yet?

**Comment by friedbunny** [comment]
Still a thing that needs to be done, stalebot.
